src/plugins/discover/public/application/main/components/top_nav/esql_dataview_transition/esql_dataview_transition_modal.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState, useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FEEDBACK_LINK } from '@kbn/esql-utils';
+import {
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiText,
+  EuiCheckbox,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiLink,
+  EuiHorizontalRule,
+} from '@elastic/eui';
+
+export interface ESQLToDataViewTransitionModalProps {
+  onClose: (dismissFlag?: boolean, needsSave?: boolean) => void;
+}
+// Needed for React.lazy
+// eslint-disable-next-line import/no-default-export
+export default function ESQLToDataViewTransitionModal({
+  onClose,
+}: ESQLToDataViewTransitionModalProps) {
+  const [dismissModalChecked, setDismissModalChecked] = useState(false);
+  const onTransitionModalDismiss = useCallback((e) => {
+    setDismissModalChecked(e.target.checked);
+  }, []);
+
+  return (
+    <EuiModal
+      onClose={() => onClose()}
+      style={{ width: 700 }}
+      data-test-subj="discover-esql-to-dataview-modal"
+    >
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          {i18n.translate('discover.esqlToDataViewTransitionModal.title', {
+            defaultMessage: 'Unsaved changes',
+          })}
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+
+      <EuiModalBody>
+        <EuiText size="m">
+          {i18n.translate('discover.esqlToDataviewTransitionModalBody', {
+            defaultMessage:
+              "Switching data views removes the current ES|QL query. Save this search to ensure you don't lose work.",
```

Comment:
```suggestion
              "Switching data views removes the current ES|QL query. Save this search to avoid losing work.",
```

Reply:
Done https://github.com/elastic/kibana/pull/188898/commits/9405305ce03c6a4bd5831c53d8b7a073cfc9d269.

Also I changed the feedback link as you suggested. 

<img width="899" alt="image" src="https://github.com/user-attachments/assets/9e6c709a-a374-4546-b78f-ae9e12433cc4">